src/components/wallet/transformRawData.tsx

```diff
@@ -0,0 +1,121 @@
+            description: `You've redeemed ${massetUnits.simple.toFixed(2)} of ${
+              mAsset.symbol
+            } with ${mappedBassets}`,
+            hash,
+            timestamp,
+          };
+        }
+        case 'RedeemMassetTransaction': {
+          const massetUnits = new BigDecimal(tx.massetUnits, mAsset.decimals);
+          return {
+            description: `You've redeemed ${massetUnits.simple.toFixed(2)} of ${
+              mAsset.symbol
+            }`,
+            hash,
+            timestamp,
+            formattedDate,
+          };
+        }
+        case 'MintMultiTransaction': {
+          const mappedBassets = tx.bassets.map(
+            basset => bAssets[basset.id].symbol,
+          );
+          const massetUnits = new BigDecimal(tx.massetUnits, mAsset.decimals);
+          return {
+            description: `You've minted ${massetUnits.simple.toFixed(2)} of ${
+              mAsset.symbol
+            } with ${mappedBassets}`,
+            hash,
+            timestamp,
+            formattedDate,
+          };
+        }
+        case 'MintSingleTransaction': {
+          const massetUnits = new BigDecimal(tx.massetUnits, mAsset.decimals);
+          const mappedBasset = bAssets[tx.basset.id].symbol;
+          return {
+            description: `You've minted ${massetUnits.simple.toFixed(2)} of ${
+              mAsset.symbol
+            } with ${mappedBasset}`,
+            hash,
+            timestamp,
+            formattedDate,
+          };
+        }
+        case 'PaidFeeTransaction': {
+          const massetUnits = new BigDecimal(tx.massetUnits, mAsset.decimals);
+          const mappedBasset = bAssets[tx.basset.id].symbol;
+          return {
+            description: `You've paid a fee of ${massetUnits.simple.toFixed(
+              2,
+            )} in ${mAsset.symbol} with ${mappedBasset}`,
+            hash,
+            timestamp,
+            formattedDate,
+          };
+        }
+        case 'SavingsContractDepositTransaction': {
+          const massetUnits = new BigDecimal(tx.amount, mAsset.decimals);
+          return {
+            description: `You've deposited ${massetUnits.simple.toFixed(
+              2,
+            )} of ${mAsset.symbol}`,
+            hash,
+            timestamp,
+            formattedDate,
+          };
+        }
+        case 'SavingsContractWithdrawTransaction': {
+          const massetUnits = new BigDecimal(tx.amount, mAsset.decimals);
+          return {
+            description: `You've withdrawn ${massetUnits.simple.toFixed(
+              2,
+            )} of ${mAsset.symbol}`,
+            hash,
+            timestamp,
+            formattedDate,
+          };
+        }
+        case 'SwapTransaction': {
+          const massetUnits = new BigDecimal(tx.massetUnits, mAsset.decimals);
+          return {
+            description: `You've swapped ${tx.inputBasset} for ${tx.outputBasset} with the value of ${massetUnits} ${mAsset.symbol} `,
```

Comment:
"You swapped XXX (input) for YYY (output)"

Also these need to be the symbols, not addresses